lib/api/src/externals/function.rs

```diff
@@ -466,15 +467,26 @@ impl Function {
         }
 
         // Call the trampoline.
-        if let Err(error) = unsafe {
-            wasmer_call_trampoline(
-                self.exported.vm_function.vmctx,
-                func.trampoline,
-                self.exported.vm_function.address,
-                values_vec.as_mut_ptr() as *mut u8,
-            )
-        } {
-            return Err(RuntimeError::from_trap(error));
+        if trampoline_checked {
```

Comment:
Seems good for a start, we can easily implement this without having to do the check at runtime though! We can make a trait and use static dispatch to generate 2 versions of this function.

We can do that as a follow up to this PR though

Reply:
Can we have the function as `#[inline]` so the compiler does that for us?

Reply:
Yeah! Although I believe you don't need to mark functions as `#[inline]` to have them be inlined! In fact I'm not sure if `#[inline]` does anything other than document that you want it be inlined. I know `#[inline(always)]` and `#[inline(never)]` definitely do things though.

That is one of the nice things about static dispatch in situations like this: we trade binary size (i.e. we generate more code) for specialized versions of the functions that are optimized to do 1 thing.

Reply:
To clarify what I mean by using static dispatch here we can have something like:

```Rust
trait Trait {
    unsafe fn call_trampoline(..) -> ...;
}

struct Checked;
impl Trait for Checked {
   unsafe fn call_trampoline(..) -> .. {
      call_checked(..)
    }
}

struct Unchecked;
impl Trait for Unchecked {
   unsafe fn call_trampoline(..) -> .. {
      call_unchecked(..)
    }
}

fn call<T: Trait>(...) -> ... {
   // then instead of an if statement we just call
   unsafe {
      T::call_trampoline(..)
    }
}
```